### ISSUE TITLE
RM Bridge not funding hub

### ISSUE THREAD

**Issue by beckyricha** [beckyricha/Broadlink-RM-SmartThings-Alexa#17]
Over the last week or 2, several users reported purchasing a hub, but can't get the RM Bridge to find it.  In order to understand whether this is a change to the new Broadlink devices, please post if you have recently purchased one and it works fine.  Please include your model and firmware version.  At first I assumed it was something in the users' network setup, which still may be, but at least 4 people reported the issue in the last 2 weeks so something seems to be amiss.

I have also contacted the RM Bridge developer to inquire whether he is getting similar feedback.

For those who have encountered this issue, please advise whether using the Tasker app as a bridge resolved it.

For those having the issue, using the "add hub manually" button on the setup page seems to always return an error that reads "not an RM2" or something like that, for everyone.  It does this for hubs that otherwise work fine, so seems to be a more general issue with the setup interface for all RM Pro devices, and not an additional symptom of your problem.  

Please feel free to add anything here that you think may be of diagnostic value.  I also recommend only purchasing hubs with a nice return policy until we ascertain the cause and any resolution.  You may also do OK with a used hub with older firmware but that is only a guess.

**Comment by beckyricha** [comment]
Update 1-19-2017: I ordered a new hub to see if new firmware is the cause.  I'll try it Saturday and post again.

**Comment by beckyricha** [comment]
If anyone is on the fence because of wondering if it iss enww firmware, note that I was looking <a target="_blank" href="http://rover.ebay.com/rover/1/711-53200-19255-0/1?icep_ff3=9&pub=5575262058&toolid=10001&campid=5338021734&customid=&icep_uq=broadlink+rm+pro&icep_sellerId=&icep_ex_kw=&icep_sortBy=12&icep_catId=293&icep_minPrice=&icep_maxPrice=&ipn=psmain&icep_vectorid=229466&kwid=902099&mtid=824&kw=lg">at eBay</a><img style="text-decoration:none;border:0;padding:0;margin:0;" src="http://rover.ebay.com/roverimp/1/711-53200-19255-0/1?ff3=9&pub=5575262058&toolid=10001&campid=5338021734&customid=&uq=broadlink+rm+pro&mpt=[CACHEBUSTER]"> and some of their new hub prices are currently less than Amazon, but there are also very inexpensive used ones that would likely not be affected if this issue is indeed due to new firmware (I saw one today for $16, but on eBay your results may vary).

**Comment by beckyricha** [comment]
update 1/22/2017: I received a new Broadlink rm pro device and updated its firmware to v20025 (the current version). the device was discovered by rm bridge and work normally. Whatever issue people are having appears to be with their networks.  The manufacturer has not made a change that affects this program.

**Comment by iguman** [comment]
This is my firmware but I dont have update

![img_0575](https://cloud.githubusercontent.com/assets/17013189/22186883/54e4aa3c-e0fd-11e6-9991-96cf1584e6b0.PNG)

**Comment by beckyricha** [comment]
I am not assure what you are asking me.
On Jan 22, 2017 5:49 PM, iguman <notifications@github.com> wrote:This is my firmware but I dont have update

**Comment by amitchorasiya** [comment]
Thank you. I also updated firmware on my broadlink. Its working fine.

**Comment by iguman** [comment]
Do I need to enter link for install firmware or i need to wait to show in app?

**Comment by beckyricha** [comment]
I don't know. I just went to the app and selected update and that was what showed. I am not sure if it's about the firmware or not though. I only updated because the one i received had the same firmware as my other one. Because i bought it to test whether new firmware made it worse, i updated to check that.  
If you're is working now i would not bother.

On Jan 23, 2017 4:06 AM, iguman <notifications@github.com> wrote:Do I need to enter link for install firmware or i need to wait to show in app?

**Comment by beckyricha** [comment]
I came home from messing around with the new broadlink i ordered (set it up at a relative's house) only to find that the fun2code site could not find my hub.  I never encountered that before. I uninstalled the e control app and everything worked perfectly. Maybe a coincidence but please try this as a troubleshooting step if you encounter this.

**Comment by iguman** [comment]
Thank you, I will try today

**Comment by iguman** [comment]
beckyricha I find working solution. Upgrade firmware from v20024 to v20025 from link : http://cn-fwversions.ibroadlink.com/firmware/download/10026/20025.bin and everything works . Broadlink does not give me automatic update so I go on advanced tab in eControl firmware update section . Thank you for help

**Comment by hiato6** [comment]
I'm having the same issue with the same device.. MAC not rm2 error.. tried downgrading the firmware to 20022 and upgrading to 20024 and 20025 but still got the same error.. also tried different Android devices and different computers, but no luck.. any advice?

**Comment by beckyricha** [comment]
It gives that error to everyone if you try to enter a hub Mac address manually. The web page only works for automatically detecting the device.
I have a nearly complete alternate web interface under development to make it simpler to add devices compatible with this code and with the Alexa skill i am developing. I will try to get an expedited version out even if all features aren't there in the next day or 2. Maybe it will help, but i won't hold my breath because it uses the same bridge API. The key difference is we will be able to more easily see what error codes may be coming back to help troubleshoot.

On Feb 16, 2017 3:22 PM, hiato6 <notifications@github.com> wrote:I'm having the same issue with the same device.. MAC not rm2 error.. tried downgrading the firmware to 20022 and upgrading to 20024 and 20025 but still go the same error.. also tried different Android devices and different computers, but no luck.. any advice?

**Comment by hiato6** [comment]
Becky, i just managed to get it fixed.. just downgraded RM Bridge Android app to 1.3.0 instead of 1.3.1 and it immediately learned all the commands.. Please pass this through to everybody who's having the same issue. Thanks for you support!

**Comment by beckyricha** [comment]
Great. I'm glad it worked for you.
On Feb 16, 2017 4:47 PM, hiato6 <notifications@github.com> wrote:Becky, i just managed to get it fixed.. just downgraded RM Bridge Android app to 1.3.0 instead of 1.3.1 and it immediately learned all the commands.. Please pass this through to everybody who's having the same issue. Thanks for you support!

**Comment by hiato6** [comment]
Yup, me too!
Quick question.. can the cloud solution work without the RM bridge running locally on an Android device?

**Comment by beckyricha** [comment]
Nope. Sorry. The difference is not entirely local versus cloud either. The difference is whether you want the smart app to send an http command to your device directly using an externally accessible ip address and a router port open (the cloud version), or to send a command to your hub using your network info that the smartthings app already has. In the latter case (the lan version) the hub then sends the http command to your device over your LAN. I funds the latter a little slower and less reliable but by so little that it may be my imagination based on my experience writing the code.  
Both do have some cloud elements, even more so if you add Alexa, but there is a slight difference in how much you have to open up in the router and whether you need a dns service. I also felt better including a version using the hub if I was going to use the smart things service :). Everything else works the same.
I did write to broadlink to see if I can get direct access to their api to skip the Android at least in my Alexa skill, but have not heard back yet.
On Feb 16, 2017 5:45 PM, hiato6 <notifications@github.com> wrote:Yup, me too!
Quick question.. can the cloud solution work without the RM bridge running locally on an Android device?

**Comment by YairoR** [comment]
Hi,

I've just purchased the RM PRO device and I have the same problem, but I dig a little deeper.

Once I'm entering to the http://rm-bridge.fun2code.de/rm_manage/code_learning.html page and press on the 'Load Devices' button, it just searches for a few seconds and doesn't show anything in the drop list. BUT looking at fiddler/console traces, you can see that it does find my device:

**Fiddler Response**
angular.callbacks._5({"code":0,"list":[{"id":1,"key":"myKey","lock":0,"mac":"myDeviceMac","name":"myDeviceName","password":"myPassword","subdevice":0,"type":"**RM PRO**"}],"msg":"Execute success"})

**Console Trace:**
RM Bridge Devices:
{"code":0,"list":[{"id":1,"key":"myDeviceKey","lock":0,"mac":"myDeviceMac","name":"myDeviceName","password":"myDevicePassword","subdevice":0,"type":"**RM PRO**"}],"msg":"Execute success"}

So I thought maybe the reason I'm not seeing it under the devices is related to some Angular issue, so I downloaded the offline version and looked at the angular code (learn.js; line 80):

if (type == "RM1" || type == "RM2" ||
                                    type == "10001" || type == "10002" || type == "10026") {
                                    rm_devices.push(data.list[i]);
                                }

As you can see, the returned type for my device is **RM PRO** but looks like the rm-bridge web doesn't support it.

Does it might be related to the e-control app version?

**Comment by beckyricha** [comment]
The bridge web interface supports it the RM pro fine.  Make sure you have updated your hub's firmware.
Otherwise you can use the alternate web interface i plan to release this weekend.

On Feb 17, 2017 5:39 AM, YairoR <notifications@github.com> wrote:Hi,
I've just purchased the RM PRO device and I have the same problem, but I dig a little deeper.
Once I'm entering to the http://rm-bridge.fun2code.de/rm_manage/code_learning.html page and press on the 'Load Devices' button, it just searches for a few seconds and doesn't show anything in the drop list. BUT looking at fiddler/console traces, you can see that it does find my device:
angular.callbacks._5({"code":0,"list":[{"id":1,"key":"myKey","lock":0,"mac":"myDeviceMac","name":"myDeviceName","password":"myPassword","subdevice":0,"type":"RM PRO"}],"msg":"Execute success"})
RM Bridge Devices:
{"code":0,"list":[{"id":1,"key":"myDeviceKey","lock":0,"mac":"myDeviceMac","name":"myDeviceName","password":"myDevicePassword","subdevice":0,"type":"RM PRO"}],"msg":"Execute success"}
So I thought maybe the reason I'm not seeing it under the devices is related to some Angular issue, so I downloaded the offline version and looked at the angular code (learn.js; line 80):
if (type == "RM1" || type == "RM2" ||
type == "10001" || type == "10002" || type == "10026") {
rm_devices.push(data.list[i]);
}
As you can see, the returned type for my device is RM PRO but looks like the rm-bridge web doesn't support it.
Does it might be related to the e-control app version?

**Comment by beckyricha** [comment]
Btw i did some of that before and also tried changing the code in that line to make it OK if it returns the type rmpro but that is not enough of the issue.  My guess is when you update the hub firmware it Chevy's the issue and returns more what the bridge is expecting. 
On Feb 17, 2017 5:39 AM, YairoR <notifications@github.com> wrote:Hi,
I've just purchased the RM PRO device and I have the same problem, but I dig a little deeper.
Once I'm entering to the http://rm-bridge.fun2code.de/rm_manage/code_learning.html page and press on the 'Load Devices' button, it just searches for a few seconds and doesn't show anything in the drop list. BUT looking at fiddler/console traces, you can see that it does find my device:
angular.callbacks._5({"code":0,"list":[{"id":1,"key":"myKey","lock":0,"mac":"myDeviceMac","name":"myDeviceName","password":"myPassword","subdevice":0,"type":"RM PRO"}],"msg":"Execute success"})
RM Bridge Devices:
{"code":0,"list":[{"id":1,"key":"myDeviceKey","lock":0,"mac":"myDeviceMac","name":"myDeviceName","password":"myDevicePassword","subdevice":0,"type":"RM PRO"}],"msg":"Execute success"}
So I thought maybe the reason I'm not seeing it under the devices is related to some Angular issue, so I downloaded the offline version and looked at the angular code (learn.js; line 80):
if (type == "RM1" || type == "RM2" ||
type == "10001" || type == "10002" || type == "10026") {
rm_devices.push(data.list[i]);
}
As you can see, the returned type for my device is RM PRO but looks like the rm-bridge web doesn't support it.
Does it might be related to the e-control app version?

**Comment by abeln2672** [comment]
I'm having this problem as well.  Guess I picked the wrong time to buy a Broadlink, haha :-)

At first I couldn't get RM Bridge to find my device at all ("Load Device" would return nothing).  I tried manually entering the MAC and then "Learn Code", but got the error mentioned (MAC: XX:XX...not RM2).  I then downgraded to version RM Bridge 1.3.0 as recommended and updated my RM2 firmware to 20025.  RM Bridge now "finds" my device ("Load Device" returns the correct device name and MAC), but I get the same error when I try to Learn Code -- "Not RM 2".  

Thanks for all your hard work @beckyricha .  I'll monitor this thread for updates.  Best!

**Comment by beckyricha** [comment]
What happens if you upgrade the RM bridge? I never had to downgrade it to eliminate this error.
On Feb 18, 2017 6:56 PM, abeln2672 <notifications@github.com> wrote:I'm having this problem as well.  Guess I picked the wrong time to buy a Broadlink, haha :-)
At first I couldn't get RM Bridge to find my device at all ("Load Device" would return nothing).  I tried manually entering the MAC and then "Learn Code", but got the error mentioned (MAC: XX:XX...not RM2).  I then downgraded to version RM Bridge 1.3.0 as recommended and updated my RM2 firmware to 20025.  RM Bridge now "finds" my device ("Load Device" returns the correct device name and MAC), but I get the same error when I try to Learn Code -- "Not RM 2".
Thanks for all your hard work @beckyricha .  I'll monitor this thread for updates.  Best!

**Comment by abeln2672** [comment]
Just updated to 1.3.1 and seeing the same error message.  I only tried
downgrading because someone else in the thread mentioned that being a
solution.  Seems upgrading the device firmware got it "recognized" by RM
Bridge, so that's progress?  :-)  But stll the "not rm2" error.

Oh, and I did try another device for the web interface (my cell phone).
And the RM2 works fine with the e-control app.  Anyway, I know it's really
not your issue and you've been unable to replicate it, but I appreciate you
responding...

**Comment by YairoR** [comment]
For me upgrading the hub's firmware to 20025 did the trick. Now my hub's name is in Chinese, but I can live with that :)

**Comment by beckyricha** [comment]
I think you can rename it in the e control app. Mine in Chinese to though. To lazy to change it.
On Feb 19, 2017 2:38 AM, YairoR <notifications@github.com> wrote:For me upgrading the hub's firmware to 20025 did the trick. Now my hub's name is in Chinese, but I can live with that :)

**Comment by beckyricha** [comment]
Have you tried recording codes for different devices? Might be something with the remote code but returning the wrong error? I'm totally stumped and guessing now... 
On Feb 18, 2017 11:34 PM, abeln2672 <notifications@github.com> wrote:Just updated to 1.3.1 and seeing the same error message.  I only tried
downgrading because someone else in the thread mentioned that being a
solution.  Seems upgrading the device firmware got it "recognized" by RM
Bridge, so that's progress?  :-)  But stll the "not rm2" error.

Oh, and I did try another device for the web interface (my cell phone).
And the RM2 works fine with the e-control app.  Anyway, I know it's really
not your issue and you've been unable to replicate it, but I appreciate you
responding...

**Comment by abeln2672** [comment]
Thanks again for even guessing...I'm stumped, as are many others it would
seem.  I get that error before I even push a button on my remote.  Almost
instantly after I push "Learn Code", that "... not rm2" error pops up.
Just for kicks, I tried pushing a number of different remote buttons before
and immediately after clicking Learn Code, but same error.

Do you have any idea how to downgrade the firmware?  I got a reply in the
ST forum that a combination of either 20022 or 20024 with RM Bridge 1.3.0
had worked for some people.  Figure it's worth a shot...

Of course I eagerly anticipate trying your new web interface just to see if
that's more successful!  I subscribed to the announcements in your forum...

**Comment by beckyricha** [comment]
Tomorrow I'm off work and i will post an alternate web interface online you can try. I plan to do more with it but the outset you need works so i can get that out there. You might also take a look at the api on the fun2code web page. There's a chrome plugin that will send those post data commands easily if you don't already have something like that handy to play with.  I'll dig up the name of it when I'm home if you like.

On Feb 19, 2017 8:05 PM, abeln2672 <notifications@github.com> wrote:Thanks again for even guessing...I'm stumped, as are many others it would
seem.  I get that error before I even push a button on my remote.  Almost
instantly after I push "Learn Code", that "... not rm2" error pops up.
Just for kicks, I tried pushing a number of different remote buttons before
and immediately after clicking Learn Code, but same error.

Do you have any idea how to downgrade the firmware?  I got a reply in the
ST forum that a combination of either 20022 or 20024 with RM Bridge 1.3.0
had worked for some people.  Figure it's worth a shot...

Of course I eagerly anticipate trying your new web interface just to see if
that's more successful!  I subscribed to the announcements in your forum...

**Comment by beckyricha** [comment]
Try this interface (link at the end).  I'm hoping this may help people having trouble getting the hubs recognized.  It's still a little rough (let's call it an alpha version...) but you don't have to remember rules for telling it if it's on or off, and it adds code editing that was not there before. You do have to use your external ip address and forward the port on your router, because it runs on a Google apps script, but it doesn't have to use a dynamic dns server.  If you need to find your address quickly, go to whatismyip.org to get it.   

If you use this to edit existing codes, make a backup of your code data first. If you can't do that you may wish to wait for a more stable version. The only way to edit with a script is to delete the old then add the new, so if something goes haywire in the middle, you may have to re-record a code.  Try it first on ones you know you can easily re-record.   

I think when finished this will have clearer feedback and be more focused for this application  than the current interface, but please let me know any suggestions or bugs you find.  I know it's weird, but the reason it uses Google is that it is ultimately going to be for an Alexa skill that piggybacks Google's authorization scheme and stores your login and ip info securely on your google drive along with info about multiple command devices and TV channels.  I know the authentication looks odd but I am not grabbing your data.  I'll be happy to make the script viewable to anyone who wants it, but if it concerns you, you can turn off your RM Bridge security temporarily and leave that blank here, then turn it back on when you finish.  It's up to you.  

After this gets working well I also plan to add multi-command device entry and TV channel entry and get the Alexa skill published.  The draft Alexa skill does work for single device on/off commands, so I will post soon for anyone with the ability and time to set up an Alexa skill that's not in the Amazon store yet.  For those who've done it, it is a bit faster with the smart home skills than the custom ones.

Anyway, here's the interface link:
https://script.google.com/macros/s/AKfycbxJd2rD8H8l2BSJ9ZWCngRZjjBqi2w1ly3jg1ovqPoOez00Mibk/exec

**Comment by abeln2672** [comment]
Can't thank you enough for all your efforts!  Things are working now, but I
didn't have to use this interface.  I'll explain what happened, but it's
not likely to help anyone in the future because I frankly have no idea why
things just started working.

Saw your post and immediately clicked the interface link.  Forwarded the
port on my router to my RM Bridge device, which I had set with a static
IP.  Clicked "Load Hubs," but got nothing.  Thought that was odd as the
original RM Bridge web interface was finding my hub previously with no
problem, but it couldn't learn codes.  So I went back and tried the old
interface, but now it couldn't even find my hub at all (404 error).
Frustrated, I reset the phone running RM Bridge.  When it restarted, it had
a new IP address (not the static one I set).  Back in the old RM Bridge
interface, I plugged in the new IP and it immediately found a hub -- but
now with a name in Chinese characters (not the generic "eRemote" name).  I
clicked "learn codes" and did NOT get the error ("not rm2")...just a nice,
inviting, spinning circle!  Proceeded to follow normal code-learning
procedures, ST device handlers, etc...and things work.

So in total, over the course of 4 days, I tried upgrading the hub firmware
to 20025 (which seemed to take), then downgrading RM Bridge to 1.3.0, then
updating it back to 1.3.1.  All of this got my hub recognized as "eRemote",
but it could not learn codes.  I then just let it sit a couple days.  When
I came back to it, it wasn't being recognized until I restarted my device
with RM Bridge.  Then the hub was recognized with a Chinese name and
learning codes worked. I'm stumped...

Thanks again for your support!  I'll still wait on the full release of your
new interface, as there are some other devices and button combos I want to
learn, but don't feel like doing under the old procedures.

**Comment by eflok** [comment]
can somebody help me how do I upgrade firmware for broadlink from v20024 to v20025. when I try to do it message coming up no upgrade found.

**Comment by febriiswanto** [comment]
did Anyone buy a new Broadlink RM Pro **(RM03 2017 version)** and successfully integrate the hub using this method already?
I have Broadlink RM Pro (old version) and it is working fine, wanna buy a new RM to place it in my another room but not sure with the new RM release sold in the market recently.

Please advice.. thanks!

**Comment by amitchorasiya** [comment]
I have RM3, it works.

**Comment by febriiswanto** [comment]
Great if it supports RM03.

Do we need to register RM03 manually or it is recognized automatically?

**Comment by beckyricha** [comment]
All of the devices I have encountered so far have been recognized when they have updated firmware. The manual entry doors but work though.  I am not sure which one is rm03. If you are asking to ensure it works before purchasing something, I would suggest getting in touch with the author of the RM bridge app (if that is the one you plan to use).  Ib write something that uses it, but the recognition guess through that app, which I can't alter.   On Aug 3, 2017 3:46 AM, febriiswanto <notifications@github.com> wrote:Great if it supports RM03.
Do we need to register RM03 manually or it is recognized automatically?

**Comment by amitchorasiya** [comment]
Becky is right. Better check with author if u have doubt.

I bought RM3 from amazon so i thought i can return if doesnt work. U didnt
find amy issues, it worked like RM2.

**Comment by febriiswanto** [comment]
Okay thanks for the suggestions. Will buy it first then work around with that.

Anyway this is the RM03 that I am talking about:
![rm03](https://user-images.githubusercontent.com/30674945/28919786-93fc71e8-7881-11e7-97ec-748fa90b352b.jpg)
![rm03 vs rm2](https://user-images.githubusercontent.com/30674945/28919791-99d3f654-7881-11e7-81f9-f39ffb6dfa09.jpg)

**Comment by amitchorasiya** [comment]
Yes, that what i have. See attached.

**Comment by amitchorasiya** [comment]
Attachment didn't come when replying to email.

![2017-08-03-07-35-31](https://user-images.githubusercontent.com/21311018/28920232-e5943756-781e-11e7-8485-ddbd70d17ba5.jpg)

**Comment by febriiswanto** [comment]
Thank you so much, confirmed already.. :)

**Comment by ilan316** [comment]
I have the RM Pro 2, 
firmware 20025

there are times when i try remotely to connect to my RM , and i get "device offline" on the e-control app
why is that and how can i make sure not to get the offline message

**Comment by adamgab05** [comment]
Hi all. I'm also having an issue adding a new RMpro to rm bridge.

What is strange is that a device is automatically detected when hitting load devices... but the mac address detected is not the broadlink device 

If I try and add manually, the only way it will add the device to the drop down menu is if I input the "type" as rm2. 

However then I get the error "Not rm2" when trying to add codes. 

Any help would be appreciated...

Oh and I have updated firmware on the broadlink

**Comment by Tanny22** [comment]
Hi everyone. I, too having the Unable to send code to eRemote error (not rm2) while trying to add command to Blackbean. 
Other than that, when I try to do the same thing on Broadlink RM pro, I got the "unable to manage this device" error. Anly solution ? 
Your help is much appreciated !

**Comment by febriiswanto** [comment]
Hi I am also having problems after bought the New 2017 RM Pro RM03 device. 
My previous two broadlink RM Pro work fine and the RM bridge could detect it, but my new RM Pro is not recognized by the RM bridge. 

I noticed that this new broadlink RM Pro has a different pattern of MAC address:
Older two RM have Mac Address > b4:43:xx:xx:xx, 
Newer RM has Mac Address > 34:ea:xx:xx:xx.

Might it possible this issue is caused by the Mac address? Anyone has knowledge about this?

Appreciate your help. 

Thanks

**Comment by frazered** [comment]
Hello,  I have the new 2017 RM Pro (Serial# starting with RM09) which works perfectly fine with the e-control and IHC app.
I have previously configured 2 RM Mini 3s with RM Bridge and the Site without any issues.

However, this new 2017 RM 3 Pro refuses to show up either by "Load Devices" or by the "Manual MAC Address" method.   I have tried entering various "Types" in the field like RM2, RM3 RM Pro etc none of them work.

Can anyone advise on a solution?  Also what is the correct "Type" that needs to be entered for these RM Pro devices.  The site seems to only accept values of "RM1" and "RM2"

**Comment by beckyricha** [comment]
In order to resolve this, please work with the author of the rm bridge app. I wrote the smart things code to link with it, but can't troubleshoot that app's device detection processes.  I am traveling right now so don't have the link to his site, but it's the same place where there code learning happens. I would recommend making sure you have installed any available updates to his app first though.Sorry I can't be of more help on this one.BeckyOn Aug 28, 2017 9:51 PM, frazered <notifications@github.com> wrote:Hello,  I have the new 2017 RM Pro (Serial# starting with RM09) which works perfectly fine with the e-control and IHC app.
I have previously configured 2 RM Mini 3s with RM Bridge without any issues.
However, this new 2017 RM 3 Pro refuses to show up either by "Load Devices" or by the "Manual MAC Address" method.   I have tried entering various "Types" in the field like RM2, RM3 RM Pro etc none of them work.

**Comment by frazered** [comment]
Thanks Becky. Have sent a note to Jochen.
Meanwhile, would you know of what "Type" should be entered in the field for RM 3 / Pro devices.
Example:- For RM Mini 3 , I entered "RM2".  What should be entered here in this field if I have a RM Pro / RM 03 device ?    (Sorry, the namings of these devices are straight out of a funhouse)

**Comment by iguman** [comment]
Do you have any information on whether the  bridge works with after rm pro firmware  update to 20028 ? With 20025 works

**Comment by beckyricha** [comment]
I am not at home to check quick which firmware mine has, but maybe someone else using this can answer.On Aug 29, 2017 9:14 PM, iguman <notifications@github.com> wrote:Do you have any information on whether the  bridge works with after rm pro firmware  update to 20028 ? With 20025 works

**Comment by EG-ITz** [comment]
> frazered commented 22 hours ago • edited
> 
> Thanks Becky. Have sent a note to Jochen.
> Meanwhile, would you know of what "Type" should be entered in the field for RM 3 / Pro devices.
> Example:- For RM Mini 3 , I entered "RM2".

Javascript used in rm-bridge checks this types: "RM1" "RM2" "10001" "10002" "10026".
I have an RM plus 2017 with serial number starting with RM09 and I can confirm you that it is not discovered by rm-bridge and none on the type code above works if you add it manually.
Probably the Broadlink SDK used by rm-bridge is not compliant with this new devices (I don't know if Broadlink will upgrade the SDK, they refused to do it for RM 3 mini)

**Comment by frazered** [comment]
Thanks teDDyGH!!  Thats a bummer. Guess will have to wait for Jochen to start supporting it.
Also, these new Broadlinks do not support 315Mhz.  There is a mega confusion among sellers and suppliers since their listings usually are for the older versions.   Only few of the shipped direct from China ones are currently supporting 315Mhz

**Comment by EG-ITz** [comment]
Consider that Rm-bridge is a facade over Broadlink SDK; if SDK does not support communication with this new versions, I doubt that Jochen can solve the situation.

**Comment by eylasa** [comment]
I found work around.
I downgraded the RM-pro to  firmware v20024 by the instruction [here](https://originallyus.freshdesk.com/support/solutions/articles/22000176674-rm-pro-firmware-v20024-not-supported).
Now both RM-plugin and RM-bridge do recognized it as RM2+ 10026. On the other hand the unit IR learning abilities dramiticlly decreased.

**Comment by frazered** [comment]
Hi eylasa,
Would you be able to share the link to the v20024 firmware? The link above takes to a page for instructions for the v20025 firmware which is the one I already have installed on the 2017 RM pro (Does not work with RM Bridge/Plugin)

**Comment by orsgah** [comment]
i have the same problem with rmpro plus 2017. manualy update to v20028, and all work fine

firmware: [here](http://cn-fwversions.ibroadlink.com/firmware/download/10026/20028.bin)

update instruction: [here](https://originallyus.freshdesk.com/support/solutions/articles/22000176674-rm-pro-firmware-v20024-not-supported)

add manualy
name: xxx
mac: xxx
type: 10026

**Comment by EG-ITz** [comment]
some more infos.
I sniffed the firmware check made by official app for my RM Plus 2017 SN starting with RM09;
This is the sniffed url:
http://fwversions.ibroadlink.com/getfwversion?devicetype=10119

So device type is 10119 and not the usual 10026. Available firmware version have the same number (20025) but  hex-comparing 20025.bin version for devicetype=10026 they are different, so I don't know if previous mentioned link, suitable for 10026 version is safe to flashing on 10119 device.
That firmware apparently solved the problem only because flashing it transformed the RM+2017 device in a 10026 device.

**Comment by ilan316** [comment]
device 10026  what is the latest firmware ?  20028?

**Comment by eylasa** [comment]
**frazered**  you are right. it is v20025 but it worked for me.

**orsgah**  I upgredad it with the link you gave. Now RM-pro call himself v20028, and work fine with RM-plugin and RM-bridge, but the IR learning ability stay poor, while it was good with previous v20028.

**So here is my suggestion how to workaround this problem**:
1.	Upgrade to V20028 by the e-control automatic upgrading method.
2.	Let it learn all the devices you want.
3.	Downgrade to v20025 using [this link](https://originallyus.freshdesk.com/support/solutions/articles/22000176674-rm-pro-firmware-v20024-not-supported).

Whenever you want to add IR device repeat these 3 stages.
This workaround doesn't help the "RM voice" users, as this app doesn't copy learned devices from e-control, but insist to learn everything alone

**Comment by frazered** [comment]
Wow. Thanks for all your insights!
I have the 2017 RM09XX and I upgraded to v20028 (through e-control) and RM Bridge Recognizes it now automatically (Device name shows up *in Chinese* when "Load Devices " is clicked)

P.S. --> Is there any way to delete old / "ghost" devices in RM Bridge? I had a RM Pro added previously but have returned it and also deleted it from the e-control and IHC app however, it still appears in RM Bridge when "Load Devices" is clicked. Any way to delete these?

**Comment by EG-ITz** [comment]
frazered, have you forced the firmware version for 10026 device?

**Comment by frazered** [comment]
I already had v20025 on my RM 2017 (10119) device so I tried to force v20025 for the 10026 device as suggested from the link above but nothing seemed to happen and RM Bridge did not work. I then rebooted device and tried RM bridge with same issue but this time, I saw v20028 magically appear in the e-control app which i downloaded and seems to have enabled RM Bridge for me.

**Comment by geekyxpert** [comment]
@frazered i have Rm pro that shows RM2+v2 (10119) is it the same as your device ? it can show in RM plug in but i always get this error with a red message Please update your device firmware in Broadlink econtrol app to be supported by this app
if yours was the same what firmware did you use ? and is it safe to flash it on 10119 devices?

@teDDyGH your device is same as mine i guess , did you manage to make it work with RM plug in ?

regards

**Comment by EG-ITz** [comment]
@geekyxpert same device code, 10119. RM bridge in my case didn't recognize my device at all, so I forced the 10026 version like @frazered and now it is available on RM Bridge and seems to work. I tested only TV remote switch, I'm not using  it every day at moment.

**Comment by geekyxpert** [comment]
@teDDyGH thank you so much for your reply, can you please tell me how did you force the update ? through the device firmware update in advanced settings? and which firmware link you used? and did you notice any performance difference in IR and RF?

**Comment by geekyxpert** [comment]
i updated it to 20028 and now its recognized on RM plugin and added the devices but still Alexa can't find the new devices !! any help or recommendation please??

**Comment by bgarg68** [comment]
Hi,
My RM2 upgraded to v30 two days back because of that RM plugin bridge between Amazon echo dot collapsed. Even RM plugin v2.0.2 updated to v2.2.0 resulting not all of my devices are being recognised by RM plugin.
Any work around would be appreciated.

**Comment by bgarg68** [comment]
Hi, 
Just an update to my earlier comment: latest RM2 v30 is seems to be working fine. All codes read by RM plugin. But now the problem is with RM Plugin v2.2.0 which is not read/handshake by Alexa. Now I don't have access to v2.0.2 of RM plugin.

**Comment by skavan** [comment]
So my 2c. Just to reaffirm some of the approaches above.

- I have a new RM pro with s/n beginning RM09.... and MAC Address beginning 34:..., as @geekyxpert 
 mentioned, this is "model #" 10119. RM Bridge doesn't support it.
- To get it to work, one needs to cross-grade it to a 10026. I used the firmware provided by @orsgah, above and I am now on f/w 20028 - but this is the 10026 version.
- RM Bridge now recognizes it properly.

Now, I don't know whether this is the best f/w to be on -- perhaps someone can comment. But until RM Bridge adds support for the 10119 model, we should maintain a set of links for the 10026 firmware files and perhaps people can comment on which ones work best.

Hope it helps.

**Comment by websterjazz** [comment]
@geekyxpert I am facing the same issue... Did you find a solution?

**Comment by lokesh2404** [comment]
I am completely new to these things...recently got Echo and RM Pro3. Thought adding RMPro to Econtrol or IHC will be easy but it never worked. In Broadlink app RMpro gets configured but is not visible in the list at all. In IHC app it fails always. 
I understand that firmware upgrade or downgrade could be done only if you can see device in the list. So I cannot even do that. 
Any help?. I tried playing with wifi router settings but no luck. 
Please help

**Comment by adamgab05** [comment]
I found a solution. I went out and got a harmony hub.

No RF. But it turns on the tv and integrates out of the box with Google and
Alexa.

**Comment by lokesh2404** [comment]
finally got the device be visible in IHC n Broadlink app...now quest to link with Alexa begins

**Comment by hiato6** [comment]
Can you share how you fixed it?

**Comment by lokesh2404** [comment]
2 things worked for me - 
1) As my wifi was not connecting, I tried using phone hotspot. it worked just fine. Both Econtrol & IHC app added RM Pro instantly. 
2) Phone hot spot was not the permanent solution,so I had to configure my wifi settings. After playing with for 3 days, I got to know that - (a) try 2 or 3 different security protocol - WPA or WEP. WPA worked for me and (b) SSID isloation was "ON" by default. I checked it off. Also keep wifi name very simple without special characters. 
Now I will try linking it to Alexa.

**Comment by MachApple** [comment]
Hi.

RM2 Pro+ is Firmware update failed.

It does not start.

Is there any recovery?

Thank you.

**Comment by Parajedic** [comment]
Hi All, I wonder if you can help at all with a problem I seem to have. I have an RM Pro with SN: beginning RM0922. Im trying to use it with IHC but it seems to not recognise RF codes as a suitable option for TV on/off. Initially it would allow it but then seemed to stop after a firware upgrade. It's currently running the latest version of e-control which prevents you from downgrading the firmware. I just want to integrate alexa with RF without having to use a hub which apparently you can do by using RF controllers for TV commands. Any help please. Thanks

**Comment by Parajedic** [comment]
Actually, all sorted further to firmware upgrade. Thank you anyways!!

**Comment by CKY1234** [comment]
I have found the solution to connect the RM pro 2017 to RM bridge app. Just use RM bridge app 1.3.0 (google it) and the RM pro connects.

Only problem I am having now, is how to learn RF codes for the TC2 switches. 
Does anyone know how I can have this set up so I can use ifttt to control the TC2 switches via Google Home?

**Comment by hiato6** [comment]
Is there a version of the RM controller (either US or UK) that only
supports RF 433MHz frequency and not 315MHz?

**Comment by Parajedic** [comment]
I don't think so. I found that you can now control RF / IR with the RM Pro using Alexa voice commands without the need for the hub.
Kind Regards
Matt  

    On Sunday, 3 December 2017, 20:45, hiato6 <notifications@github.com> wrote:
 

 Is there a version of the RM controller (either US or UK) that only
supports RF 433MHz frequency and not 315MHz?

**Comment by CKY1234** [comment]
I got it working with the Broadlink rm pro by using Rm plugin app and rm bride app. With the rm plugin app running, you can use ifttt and web hooks to send the commands to the rm pro. The rm plugin shows the rf and ir codes that can be used. Works great!

**Comment by CKY1234** [comment]
Use the rm plugin app and not the rm bridge app

**Comment by Parajedic** [comment]
I'll look into those thanks. Do you need a dedicated android device running for it. At the momemt the system runs on its own without the need for a dedicated device. 

Matt 
 
  On Sun, 3 Dec 2017 at 20:56, CKY1234<notifications@github.com> wrote:   
Use the rm plugin app and not the rm bridge app

**Comment by CKY1234** [comment]
Yes, you need a dedicated android device to run the rm plugin app. I just have it running on my android tv box.

Good luck!

**Comment by goldy1987** [comment]
my broadlink RM pro plus is in v51, and not recognised by rm plugin nor rm bridge...any help please

**Comment by Ltomi78** [comment]
Hi.
Based on advise of ez smart (https://ez-smarthome.com) developer of broadlink i have downgrade the my new rm pro+ from FW v51 to v20025. They told me to try it as my rm pro+ not workes with broadlink rm voice app....
After downgrade the firmware version i got message fw succesfully updated and current version is v0 and app shows v49 is the new version.

After that i tried to connect the device to the wifi but not works. I tried reset but no blue led and no light at all it seems it dead or brick. Is it possible some how to reboot it or reflash the firmware. I have ftdi and arduino nano cam i use it to reflash the FW.
Is it possible that the developer suggestion to downgrade it bricked the Rm pro + device?
Thank s for help please

**Comment by jakaboers** [comment]
v51 for me too (I can't even follow how many versions this device has), I was thinking about downgrading, I guess that's not the way
this is really annoying

**Comment by Ltomi78** [comment]
Yes it is.. It workes fine with IHC app but i wanted to use with voice control. I purchased the pro voice control app. After suggestion i tried the downgrade to reach rm bridge and the result:
https://youtu.be/awFZFkFeYGE
A seemingly dead device..i know what a non compatible FW can do with a HW ..but never thought i got wrong fw from an official link.

**Comment by Ltomi78** [comment]
Is there any oprtion on the pcb of broadlink to flash with serial device? Serial port to flash bootloader or fw?

**Comment by beckyricha** [comment]
Don't know as it never opened it. Would be interesting to see if anyone succeeds, as the hardware is more interesting than their locked down firmware.On Dec 12, 2017 3:00 PM, Ltomi78 <notifications@github.com> wrote:Is there any oprtion on the pcb of broadlink to flash with serial device? Serial port to flash bootloader or fw?

**Comment by Ltomi78** [comment]
![20171212_005440](https://user-images.githubusercontent.com/34459398/33911412-8e26bbb6-df92-11e7-8b65-7d245fff2dc2.jpg)
I ve checked inside (see attached pic) find 3 pin holes near thw wifi module marked with G R T. I m wondering it can be ground Tx RX ?
It would be good to know if it is possible to reflash it with ftdi or other serial tool

**Comment by Ltomi78** [comment]
I ve got the reply from the broadlink manufacturer
[
![screenshot_2017-12-13-11-20-26](https://user-images.githubusercontent.com/34459398/33933993-df5f6a54-dff7-11e7-9737-4b402252a7c6.png)
](url)

**Comment by Ltomi78** [comment]
Anybody can check that this v3.3 broadlink an oem pcb?
I v not found picture only 1.1 version opened so dont know this 3.3v pcb which i have bought is the newest or clone?
![20171212_005440](https://user-images.githubusercontent.com/34459398/33976897-4dc68afe-e098-11e7-8bb2-b20df9380b8d.jpg)
![20171211_235121](https://user-images.githubusercontent.com/34459398/33976898-4df38234-e098-11e7-881a-fa31e76a1c74.jpg)

Anybody can check that this v3.3 broadlink an oem pcb?
![2017-broadlink-rm-pro-rm03-smart-home-automation-wifi-ir-rf-universal-intelligent-remote-control-switch](https://user-images.githubusercontent.com/34459398/33976880-34cea82e-e098-11e7-961e-cf52a9e36203.jpg)

**Comment by Ltomi78** [comment]
Here is the manufacturer comment...warning before upgrade!
Dear Customer,
Pls kindly note the device of RM pro and RM pro+ are have the different firmware version, and not support upgrade for each other, when you do that, the device will die, and never work again, and also don't have any other function to help make it work again.

For the other hand, the RM plugin is not BroadLink app, so we don't think you can make it work with Alexa through the firmware version updating.

I have checked that the seller is one of our Agent, but i don't know they why advise you updating the firmware version for the device, so the device have been broken, i can't help you solve this, but i think the Agent need help you solve, thanks!



Best Regards,
International Customer Service Center
Hangzhou Gubei Electronics

**Comment by jakaboers** [comment]
Hello

That's a shame
I also received some information from the developer of RM bridge that he asked for the SDK of the new version from Broadlink but didn't get it, so he can't make it work.
Maybe we can both (or more) write to Broadlink to release SDK/API. Did you use support at ibroadlink.com address to contact them?

PS: Tomi én is magyar vagyok, ha esetleg tudok segíteni, szívesen. Szét nem szedném  feltétlenül az eszközt, mert ugye még garis. Nem gondoltál rá hogy gariba visszaküldöd, nem biztos hogy rájönnek, hogy brickelve lett, nagyjából még működik ha jól értem
Egyébként milyen flashelhető IC van a PCBn? Esetleg egy jó darabból ha lehet teljes dumpot készíteni és azt ráflashelni a tiedre, megoldódhat talán

**Comment by Ltomi78** [comment]
Hi.
Yes i have contactes three time directly to ibroadlink support. They give me very few info about FW and official upgrade also they dont know how to make it work with alexa. It is little bit strange for me. 
It s a good idea to force and email to broadlink to provide SDK/API for the rm bridge.
I m quite newbie on this field...just looking for the forums to find info but there is only few or nothing about RM pro+ ony about RM pro which i think was a 2016 release. Those devices with older Fw and hardware worka with the bridge.

PS.: Koszi higy irtal lehet hogy tudnal segiteni..esetleg messengeren elerlek valamilyen neven?
:)
A leirassal es videokkal meggyoztem az eladot egyebkent es mivel 7 napon belul volt DOA kent ugy nez ki reundaljak a az ar 95% kat.
Egyebkent igy belegondolva a pro plus supportja nagyan gyenge..sehol semmi az upgradrol es kompatibilitasrol...nem beszelve a hulye tanacsokkat ado fejelsztokrol :)

**Comment by ghost** [comment]
Has anybody got the new RM PRO+ to work. It starts with S/N RM3322.
I'm running firmware V52 and 1.3.2 in the bridge app.
YIPPEE.
Cracked it.
This is what I entered manually.
![image](https://user-images.githubusercontent.com/17054678/34454103-173d3e5e-ed5b-11e7-9f51-86a751d7c734.png)
Spoke to soon. It says it's not an RM1 or RM2 when I try learning a code.

**Comment by Bruce0286** [comment]
As coincidence I've just received the same PRO Plus serial starting RM3322 running firmware V52. Did you have any success getting this to work @bobblesg?

**Comment by ghost** [comment]
I'm afraid not.
I emailed the developer of the Broadlink Bridge website and this was his reply.
No joy I'm afraid. Going to send mine back and see if I can get my hands on an older version.

**Hi Bob,
unfortunately there is no workaround.
I tried to get the latest SDK from BroadLink multiple times without success.
The last response some days ago:
"Sorry, we have no any API for overseas market now."

So unfortunately I cannot provide any updates for new devices.**

**Comment by Bruce0286** [comment]
Bob,

Thanks for the update, I did email the developer myself this morning also. I have just let him know you have updated me. Thanks for letting me know. I might just pay the extra and invest in a harmony.

It’s a shame, if the Alexa skill was available outside the US this wouldn’t be an issue! Ah well!

Thanks again.

**Comment by goldy1987** [comment]
RM Pro plus, firmware V52, working only with IHC app.

**Comment by Ltomi78** [comment]
Mine was worked with v51 and eControl app.
V52 fw does not work with eControl only with IHC?

**Comment by Papabear73** [comment]
I also have PRO+ with serial no RM3322...
It works fine with both eControl and IHC (v52). However I can't find a way make it work with RM Bridge.
I set it up as type 10026 (instead of RM02) on the rm-bridge.fun2code.de site, it seems it comes with both "Frequency Scan" and "Learn Code" buttons. However I receive this when pressing "Learn Code:
Error learning code: mac:34:ea:34:xx:xx:xx not Rmpro+
Similar message when pressing Frequency Scan:
mac:34:ea:34:xx:xx:xx not Rmpro+
(I replaced the MAC address with x)
Basically RM Bridge keeps telling me that my device is not an Rmpro+ device. 
I tried with both RM Brigde versions: 1.3.0. and 1.3.2
Just one addition: RM Bridge keeps a cache of the devices added manually on the fun2code website. You should stop the app, clear the cache on the android to get it forgotten. 
p.s. szintén magyar

**Comment by beckyricha** [comment]
I was also working on an alternate web site for learning RM bridge codes more easily for specific use with this device handler (i.e. It adds the "on" and "off" for you and simplifies some other steps such as deleting and editing device names) but did not get any feedback and did not polish it.  Might be worth a try in case it is something hard coded into the fun2code web site.  However as I look around GitHub I'm not finding it. I may need to re post that page when I get home later.  I will add a link here soon.BeckyOn Jan 2, 2018 6:03 PM, Papabear73 <notifications@github.com> wrote:I also have PRO+ with serial no RM3322...
It works fine with both eControl and IHC. However I can't find a way make it work with RM Bridge.
I set it up as type 10026 (instead of RM02) on the rm-bridge.fun2code.de site, it seems it comes with both "Frequency Scan" and "Learn Code" buttons. However I receive this when pressing "Learn Code:
Error learning code: mac:34:ea:34:xx:xx:xx not Rmpro+
Similar message when pressing Frequency Scan:
mac:34:ea:34:xx:xx:xx not Rmpro+
(I replaced the MAC address with x)
Basically RM Bridge keeps telling me that my device is not an Rmpro+ device.
I tried with both RM Brigde versions: 1.3.0. and 1.3.2

**Comment by maxphotos** [comment]
I have the same problems.
RM PRO+ just bought. I had firmware 051 and then updated to 052.
IHC and e-Control they both work fine. The RM PRO+ is working correctly doing everything has to do.
The only problem is that I cannot set up the RM plugin I just bought for Tasker: it cannot find the RM PRO+ from the e-Control application (as stated in the instructions of RM plug in).
Any idea or fix ?
Thanks
Max from Italy

**Comment by shiftyshew** [comment]
I am having exactly the same problem as maxphotos above, written to the support and absolutely no response. I am not a programmer or software expert - total novice who just wants Google Home mini to talk to RM Pro+ via RM Plugin..... will keep watching with interest and advise if anything positive comes back... any advise on other options most grateful... :-)

**Comment by Ltomi78** [comment]
It seems we need ti buy an older device. Check this:
https://youtu.be/FWRrQ-_uwEs

What do you think which shop sell the older pro version.  ? BG?

**Comment by glynndrago** [comment]
Same problems here. I just got my RM Pro Plus and couldn`t get either RM plugin or RM Bridge to work ... Tried everything when I finally stumbled across this post  and realized it wasn`t me !

Hope someone would find a workaround cause I bought this to be able to connect it to my google home via IFTT ... Will the BlackBean still work at least ?

**Comment by doridave** [comment]
So glad I found this. I too have the new version (RM Pro+) Firmware v52 and couldn't get my head around why it works with the e-Control but all other reverse engineered tools were not. I have 2 older devices I've hacked for my needs using this and other GitHub libraries and this one was frustrating me to no end, but now at least know why.

**Comment by Ltomi78** [comment]
Hi doridave.
Where and when did you buy the rm pro plus.?
Im asking as im very confused..most of the shops provide option before buy for different rm versions..rm2, rm pro and rm pro+.
Usually the stock info says rm pro+ not in stock or discontinued but 2 month ago i could buy the rm pro+.(Unfortunatley that time i did not know about this issue with rmpro+)
Now i would like to buy the older pro version but based on review photos e.g banggood : it seems they sold rm pro+ and also simple rm pro versions in january 2018. 
So do you have any information or others here at github..where to buy an older rm pro version.?

**Comment by geekyxpert** [comment]
Hello everyone, 
An individual seller is offering me old versions of RM pro for $60 each .  I want 4 of them . Is it a good price to go for it or is it over priced ??

**Comment by Ltomi78** [comment]
Highly overpriced. Probably within a short time broadlink will provide API for developers to update rm plugin and rm bridge to work with rm pro + version also. I think its not worth to buy on twice double price.
As it seems banggood and some sellers at ali provide the older version between 32-35 usd. Im contacting some of them and ask confirmation about available versions.
If you see the reviews at BG at broadlink rm pro product you can see that some buyers got simple older rm pro version not the rm pro + in january 2018. You can see the differnce at package ..the rm pro has an iphone printed on the box. I m sure those model version has a serial begins with rm09.

**Comment by GungHo63** [comment]
RM Pro+ Same problem !
Want to connect it to smartthings

Should have bought a Harmony hub :)

**Comment by doridave** [comment]
@Ltomi78 

I bought my older models between 1-2 years ago from AliExpress. The new RM Pro+ was from Ebay a fortnight ago, here in Australia listed as "Broadlink RM3 PRO Original Home Remote Control WIFI+RF+IR".  I just wasn't aware the newer models didn't play nice with this nor davorf's nor mjg59's github repos.  My other Blackbean's both work great with OpenHab with davorf's code. I also have mjg59's working learning / sending IR codes with this RM Pro+ but nothing on RF which is why I bought it... Someone much smarter than I will come up with something, I have faith in the internet! :)

**Comment by ghost** [comment]
Probably not for this thread but does anyone know of anything else that can be used in the same way the Broadlink does. I'm after something that can learn my curtains remote and then get integrated into ST for automation.

**Comment by Ltomi78** [comment]
I can see this on all Aliexpress offer:
In this order, this the newest updated version RM-PRO+, Which is the updated version from RM-Pro. It had added the hanging board and the more stable chip to make it work better. No matter RM-Pro or RM-Pro+, they all have IR RF remote control learning control function. Besides, wifi remote control still will make your daily life more simple and convenient

**Comment by beckyricha** [comment]
I agree. For $30 I could overcome the device's shortcomings and the fact that my first one died in a few months. They are cheap Chinese electronics. Fun, but still not a $60 device.  Depending on what you want to control you could pick up a hook for much less on Amazon. Those are much sleeker and have an open interface but I found them not to control as many devices. However I was an early adopter getting one through their Kickstarter, and put mine in the drawer when I found the broadlink.  They may have fixed a lot of the issues by now. It's at hooksmarthome.com and you can search the reviews to see if it's worth it to you.BeckyOn Jan 15, 2018 4:51 AM, Ltomi78 <notifications@github.com> wrote:Highly overpriced. Probably within a short time broadlink will provide API for developers to update rm plugin and rm bridge to work with rm pro + version also. I think its not worth to buy on twice double price.
As it seems banggood and some sellers at ali provide the older version between 32-35 usd. Im contacting some of them and ask confirmation about available versions.
If you see the reviews at BG at broadlink rm pro product you can see that some buyers got simple older rm pro version not the rm pro + in january 2018. You can see the differnce at package ..the rm pro has an iphone printed on the box. I m sure those model version has a serial begins with rm09.

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.


  
  




On Jan 15, 2018 4:51 AM, Ltomi78 <notifications@github.com> wrote:Highly overpriced. Probably within a short time broadlink will provide API for developers to update rm plugin and rm bridge to work with rm pro + version also. I think its not worth to buy on twice double price.
As it seems banggood and some sellers at ali provide the older version between 32-35 usd. Im contacting some of them and ask confirmation about available versions.
If you see the reviews at BG at broadlink rm pro product you can see that some buyers got simple older rm pro version not the rm pro + in january 2018. You can see the differnce at package ..the rm pro has an iphone printed on the box. I m sure those model version has a serial begins with rm09.

**Comment by Ltomi78** [comment]
I ve got some promising info from BG. They confirm they had the pro version for sale but it worth to note or ask before purchase.
The other news that EZ smarthome (developer of Rm pro voice control apk)also developing a new app for direct voice control for broadlink devices. I think it means there will be no need for rm bridge to connect to the app with rm pro or rm pro+.

**Comment by Ltomi78** [comment]
I ve got some promising info from BG. They confirm they had the pro version for sale but it worth to note or ask before purchase.
The other news that EZ smarthome (developer of Rm pro voice control apk)also developing a new app for direct voice control for broadlink devices. I think it means there will be no need for rm bridge to connect to the app with rm pro or rm pro+.

This is the reply today from broadlink :
Dear Customer,
Sorry, i don't know where can buy the RM PRO with the serial RM09, and the RM plugin is not BroadLink device, so i also don't why this not support to work with RM pro+, thanks!

Best Regards,
International Customer Service Center
Hangzhou Gubei Electronics Technology Co., Ltd.

**Comment by beckyricha** [comment]
They already have such an app but it isn't great. You have to force it because it wants everything to be a tv. Just record whatever you want and call it a tv power switch. Then you can say Alexa, turn on/off...  That is built into the basic app that comes with the device. It won't work with smart things, and I have no idea about Google home, but it does work with Alexa. It is useful if you want to actually control a tv, because they have more media controls on there, not just the rudimentary on/off that was all Alexa could handle when I wrote this script.BeckyOn Jan 16, 2018 4:19 AM, Ltomi78 <notifications@github.com> wrote:I ve got some promising info from BG. They confirm they had the pro version for sale but it worth to note or ask before purchase.
The other news that EZ smarthome (developer of Rm pro voice control apk)also developing a new app for direct voice control for broadlink devices. I think it means there will be no need for rm bridge to connect to the app with rm pro or rm pro+.

**Comment by Ltomi78** [comment]
I v got my previous order from GB which i ve placed in dec. Unfortunately i did not know at dec about this osssue with pro+.
Now i have a bricked and a new pro+ device.  Im still planning to order the older version from BG but also waiting some good news about solution for the new pro+ HW and firmware. Hopefully within a few weeks broadlink release a new API for the developers of rm plugin and rm bridge...

**Comment by glynndrago** [comment]
Hi there. Gearbest do still sell the RM2 Pro ... Will this work with RM plugin ?

Also has anyone figured out how to re-flash one of these units ... I managed to brick my RM pro plus and manufacturer will not help.

On a positive note I got the RM mini 3 and it works perfectly with RM plugin for a 3rd of the price of RM pro ... Only downside is it does not control of signals.

**Comment by Ltomi78** [comment]
Hi glynndrago. Unfortunately based on manufacturer reply there is no way to reflash it...i cannot beleive this answer but also not found solution yet, furthermore we would need a ROM and a bootloder file and firmware. As i rememember the device is not recognised by the computer..? Mine also bricked by v20025 fw.

On the other hand i step forward and bought a google home mini (was only 50usd new but secondhand source in my country i dont pay shipping)which is able almost fully control rm pro plus by voice commands...you only need to make "lamp" devices and/or scenes within IHC app. See paul hibbery video on youtube.
More interesting after setup the devicea at google home app...after i can switch off the home mini or home device and im able to comtrol ihc app and rm pro plus by voice through the google home app using my android phone google assistant..so no more google hardware (mini or home)device is needed.
Od course you can use the hardware if you want control broadlink without your phone.

**Comment by glynndrago** [comment]
Can u actually control RM pro plus via Google home using ihc ?? I thought it was limited to Alexa ??? Could u please send me a link of the related video ?

**Comment by jakaboers** [comment]
I can confirm, you can, even with the newest device
Search for Paul Hibbert on YouTube.

You can use Lamp devices in IHC, which will directly show up in Google Home
App, but Lamp can only learn IR

You can use user defined to learn RF then set up scenes with that and
scenes also show up in Google

Scene and Lamp names are what you have to say to google to trigger. There
are some keywords that you should avoid, eg using light in a name might
trigger all lights etc
Shortcuts in Google are pretty useful too

On Jan 25, 2018 10:40, "glynndrago" <notifications@github.com> wrote:

Can u actually control RM pro plus via Google home using ihc ?? I thought
it was limited to Alexa ??? Could u please send me a link of the related
video ?

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/beckyricha/Broadlink-RM-SmartThings-Alexa/issues/17#issuecomment-360413419>,
or mute the thread

**Comment by Ltomi78** [comment]
glynndrago!: https://www.youtube.com/watch?v=RoJHFzh971U
You can find here the tutorial video with useful comments. I ve bought a new rm pro plus so i could test is and it works

**Comment by glynndrago** [comment]
Thanks ... Works great! 

So basically you say "Switch on/off" to trigger a "light" and "activate" to trigger a scene. You can program almost any thing without the need of an Android bridge!

**Comment by Ltomi78** [comment]
Yes sure! You can say also "turn on" and say the predefined lamp name..and within the scene you can setup universal remotes also..so RF devices also work here.

**Comment by Ltomi78** [comment]
Anyone has information how to reflash a bricked rm pro+? I also need a FW..but i also have a working device ..is it possible to readback the rom?

**Comment by glynndrago** [comment]
Same problem here ... Bricked mine updating with the wrong firmware when I was trying to get RM plugin working. Although I now have an RM mini it would be great to get the pro plus working again:-)

**Comment by Ltomi78** [comment]
I think we can start on this way to flash bricked devices as it seems broadlink rm pro and plus has the same chipset.

https://www.open-homeautomation.com/tag/wireless/
http://image.kjvkgh.com/2014/07/IMG_0513.jpg

**Comment by Buckwallen** [comment]
Hi,

New to the forum. I have created scenes and devices in ihc including lamp. All work through the app perfectly. After linking to google home, google recognises the phrase but rm is not firing for scenes. I mainly need rf to turn on power points. I noticed in Paul Hibberts vid you should be able to throw any ir or rf signal when teaching a lamp. Only ir is noticed when i perform this to the rm. Any thoughts as to why scenes arnt activating even though google home can see them and how to get a lamp appliance to recognise an rf signal when teaching.

Cheers
Pete

**Comment by Ltomi78** [comment]
Hi.
Pete you can setup universal remote to teach rf signal..than you shoul add it to a scene in ihc...than google will see that scenes and can trigger the rf remote you previously put into that scene.
I had same not want to trigger issue...i completelely resete my Rm pro amd set it up again. Also say "sync my devices" for google

**Comment by Ltomi78** [comment]
Teaching lamp as rf device works only with older rm pro versions...i n case rm pro+ with fw v51 or v52 it does not work as we know.
I have rm pro+ so i put rf remotes into scenes to make it recogniseable for home app.

**Comment by Buckwallen** [comment]
Thank you so much for your comments Ltomi.
So after testing more here is my issue exactly. 
After creating scene with ir devices and rf universal devices, I can press the scene button in the ihc app and it will work without any issues. As soon as I link broadlink smart home to Google Home, I can still activate the scenes with voice through Google Home but the RM Pro plus just wont respond. Scene still works correctly when physically pressing the button in ihc app. I have also created a scene without rf commands and still nothing after linking to Google. I have also tried a scene using an rm mini. Worked perfectly after linking to Google. My problem seems to relate directly to rm pro plus. It just wont respond after linking. It won't accept an rf frequency when teaching lamp either. I have also rebooted and factory reset. No change.

Cheers
Pete

**Comment by Buckwallen** [comment]
Sorry Ltomi. Also just realised from your message that only older rm pro rf is recognised by lamp.

**Comment by Ltomi78** [comment]
Hi Pete.
I had exactly the same issue but after reseting and setup the devices (home min and rm pro plus) again from the begining...the rm pro+ started to work with scenes also.
Previously before i ve been known about this issue i updated the firmware through ihc app from v51 to v52.
So my pro+ running with v52....i dont think it will solve your issue but you can try it..also try to say "sync my devices" command for google assistant.
So my maim solution for this not triggering issue...was to reset home mini and rm pro plus completely

In case of.lamp and rf..it is my hypotesis..as in paul hibbert video we see he can learn an rf remote as a lamp...paul has older pro he not use pro+ currently...so i think it is not a function in rm pro + but using scenes you can use rf remotes with pro+ also.

**Comment by Papabear73** [comment]
Thank you, Tomi for all the descriptions provided. I have my brand new (december, 2017) RM Pro+ (v52) working fine. As you described, devices set up as lamps can only learn IR signals, this is perfect to control anything like TV, amplifier, media player, set-top-box, even the aircon. I have LED stripes all around the room and motorized shades all around the apartment both working with RF signals. This was the primary reason I purchased Broadlink. 
Setting such devices up as a user defined appliance (with practically unlimited buttons), buttons can learn both IR and RF signals. Don't worry about the arrangement or the name of the buttons, just use names you can recognize. Setting up Scenes, you have to be careful about naming conventions. 
So all the lamps and scenes defined here will be recognized by google home. Just use the word 'activate' than the name of the scene, and there you go. 
I also had exactly the same problem, set up a few buttons, everything worked, even google home recognized them, the broadlink just did not transmit the signals. Neither IR, nor RF. Then, (thank you Tomi again), I made a full reset on the Broadlink, deleted the cache and all the data in the app, deleted and reinstalled the app, everything was started from scratch, but now works perfectly. 
And the BIG thing is that you do not need ANY Android thing powered on 24/7, no RM Bridge or RM plugin is needed, you don't even need a Google home device (it is just very comfortable to have one). 
One issue I noticed is sometimes Broadlink disconnects itself (blue led blinking slowly) no matter how close it is to my wifi router. In such a case googe home answer is correct, but nothing happens. It solves itself in a few minutes but.

**Comment by Papabear73** [comment]
How did you manage TV on/off? If set as a lamp, "switch off all the lamps in the living room" command switches on the TV. The TV obviously have the same button to switch it off and on. I only found a workaround to set it up as a scene, but then I have to say "activate my big TV" for both switching on and off. This is something I can live perfectly with, just wanted to know if you have a better solution.

**Comment by Ltomi78** [comment]
Yes..i heard about the disconnect issue..jakabeors member here facing with this issue he wrote about it some posts ago. As i know it device specific issue and probably caused by the router..but nor sure.
I think jakabeors will reply for you.

I had an other issue with wifi..but it was caused by google home as it is overpush certain type/brand wifi router with data packs make the home wifid disconnecting from the internet.
Its a known but new issue with google home devices...sometimes kill your home wifi ...and need a restart on the router.

I got a suggestion from my IT expert collegues to use an other router connected to my main router...work as a "slave" wifi...and it is dedicated for google home mini. It is a tp link router but running an opensource firmware...(i will check the name).
And NOW my mini cannot overpush my main rourer as it connects to the slave tp link router through wifi...and both the main wifi and slave remains stable
Probably you can try this in case of the broadlink ...i think its enough an other router no custom firmware necessary...but i will tell you also the type of fw i use on the slave router

**Comment by Ltomi78** [comment]
Papabear..i have no better idea than yours ;)
My scenes and tv scene looks like this i attached..i also use the turn on or activate tv both for off and on the tv..and the set top box.
Maybe you can make a separate scene for OFF commands :) dont know yet

> ..
![screenshot_2018-01-28-12-03-51](https://user-images.githubusercontent.com/34459398/35481449-0b4eb1f6-0424-11e8-88cd-ad591b5f3f6f.png)
![screenshot_2018-01-28-12-04-09](https://user-images.githubusercontent.com/34459398/35481450-0b73f16e-0424-11e8-8b8f-827a79608ecb.png)

i

**Comment by Papabear73** [comment]
Thank you Tomi, changing or adding another router is not an option for me. The problem affects all the devices with casting capabilities, and I have a lot of them in the apartment. 3 google home, 2 chromecast, 2 audicast, an android phone and tablet.... Anyway, it is supposed to be corrected by now.

**Comment by AC6y86** [comment]
Anyone find a way to make RF or scenes work with the Amazon echo?

**Comment by GraemeRadtke** [comment]
Hi Tomi, please will you explain a bit more about your setup. 
How do you connect your main router and Slave router?
Does all internet traffic go through your main router?
Is the "slave" router set up with a completely different SSID?  If so, then do you have to connect your phone to that same network to be able to configure Google Home from your phone?  
Do all your IoT devices then connect to that "slave" router by using that SSID?

**Comment by Ltomi78** [comment]
Hi!
The setup was made by my friend who working on IT field. Its not a difficult setup but i have to ask the setup he made within the router (also about the internet traffic), furhtermore that tp link router got a new non-official firmware to be able to handle large packets form googlle home mini.
My main router it is that i got from the internet service provider (modem and wifi combo), and we connected a TP-link router through LAN (with ehternet cable) port to this main router. 
Tp link has a different SSID than the main router. Only the google home mini connects to this slave tp-link router. All other devices IP camearas ,securrity system and also the broadlink connects to the main router.
Now it seems the home mini doeas not kill the internet behind the wifi of the main router as previously done when it was also connected to the main router.

Edit: so my tp link router: TL-WR1043ND , it is connected behind the main router on LAN 
It was flashed with OPEN Wrt firmware
infos here: https://lede-project.org/start
_"LEDE is designed by network professionals and others who care about the performance of their network. LEDE incorporates many algorithms from recent research that perform far better than vendor-supplied firmware. 
• LEDE is stable, and operates reliably for long periods of time.
• LEDE reduces latency/lag and increased network throughput via bufferbloat control algorithms.
• Wi-Fi improvements support higher data rates and airtime fairness between stations.
• Older devices are still supported by LEDE improvements (as long as RAM/Flash of your device can accommodate new releases) long after the manufacturer stops making updates"_

**Comment by jakaboers** [comment]
@Papabear73: as Ltomi mentioned I also have disconnect issues, I've been emailing broadlink support, they're pretty responsive but not much help really (basically try different channel, use only 20MHz bandwith, use static IP- which I can't, my router is very basic, etc).
To tell you the truth I haven't checked the LED, but I will when it happens again.

I usually have to restart the device by disconnecting power for a bit to have it reconnect. Sometimes it reconnects, but even then it takes a long time. My suspicion is that they didn't implement some kind of forced reconnection periodically at least not often enough. 
Mine is pretty close to the Wifi router, but the router shows quite low RSSI compared to other devices (-53dBm vs -45), maybe the Broadlinks wifi antenna is weak or I have to rotate it. Still it should reconnect if there is a problem.
Is this the same with yours?

**Comment by Papabear73** [comment]
Hi, I have a 3 router setup. Broadlink was just 2 m away from the main wifi when disconnected, so it is not a signal strength problem for sure. 
My router from the provider is set up as a Bridge, not as a router, wi-fi is switched off. One cable coming from this goes to my main router (ASUS RT-AC3200), which is a 3-band router. All 3 bands (one 2.4 GHz and two 5 GHz) are set up with the very same SSID, password, authentication method. My secondary router (ASUS RT-N56U) is connected to the main router, this one is set up as an access point (not as a router), its wifi has 2 bands (one 2.4 GHz and one 5 GHz). This is also set up with the same SSID, password and authentication method. 
Networkwise all of them are in the same subnet, IP addresses are given by the main router within 192.168.8.***
This way my mobile devices can roam inside the apartment picking up the wifi that has the strongest signal without re-authenticating. 
I only found my Broadlink disconnected when I was playing with it all the day last weekend. Then it reconnected itself, I did not have to do anything. Disconnecting and connecting power helps anyway. Since then, even if it disconnected, I did not notice any problem. 
Anyway, when it was disconnected, no scene or remote did actually work from the IHC until reconnection. Also, since the weekend I put the broadlink to an elevated position (180 cm from ground) to have good visibility to all the IR devices in the living room.
It would be nice to have a version with Ethernet connection :-)

**Comment by GraemeRadtke** [comment]
Thank you very much for the information Ltomi78 and Papabear73.   
I appreciate the detailed information.  I have a second router and it happens to be a  TL-WR1043ND which I have considered flashing to the other firmware because the original firmware gave me problems so that would be a good solution for me to try.  
I really want to get this setup working smoothly and trouble-free and make some good use of the Broadlink as it has a lot of potential.

**Comment by Ltomi78** [comment]
Hi!
 For those here who has a bricked RM pro+ device, i would like to give some information. I opened the hardware and identify 2 important smd components. The one marked with red circrle is Silab transmitter chip 4463 https://www.silabs.com/documents/public/data-sheets/Si4464-63-61-60.pdf
Unfotunatelly this is not the main processor of the board. The main MCU possibly is integrated into the Broadlink wifi modul, which has ID: CMIIT-ID-2017DP1434
I found only some info here https://fccid.io/CMIIT-ID-2017DP1434 ,but no programming and pinout description. 
I think it is very similar to ESP8266 wifi modul, which has an 32bit cpu integrated. So if we have fortune we can flash or program it with Arduino tool through the Gnd, TX, RX, Vcc outputs which is marked on the board with G R T. Or we can use the pinout at the two side of the module, hopefully has the same layout as ESP8266, suggested to identify with a multimeter.
http://www.tinyosshop.com/image/cache/data/wireless/wifi/ESP8266MOD%20-2-800x800.jpg
http://support.thingplus.net/en/esp8266.html
Thats the point i need help, but i will try to test if i can burn bootloader or reach somehow the integrated 32bit microcontroller. STM32 probably?
Anybody with flashing and programming skills with ARM/STM processors or integrated wifi modules here? :)
![blrmpro_mcu](https://user-images.githubusercontent.com/34459398/35617192-712f0618-066f-11e8-9ab4-0381bc1d3940.jpg)

**Comment by Ltomi78** [comment]
Just for info. I made a teardown on the wifi module..
It has an integrated Marvell 88MW300 Mcu .
https://github.com/wfr/mrvl-88mw30x-firmware-tools
Anybody here knows how to programm? :)
![20180131_213250_hdr](https://user-images.githubusercontent.com/34459398/35653072-ad22f130-06e6-11e8-8d7c-7c672240c0dd.jpg)

**Comment by marexsc** [comment]
I have the same problem, the topic is probably too new because even by typing the module model "WM3SBS03L" there is no information, besides we need the original Broadlink v51 or v52 software.

**Comment by Ltomi78** [comment]
Today..i changed my router and now my google home not trigger again my broadlink.
I m afraid of i need to delete the device in ihc to make it working again.
Any other idea? What need to reset?

**Comment by Ltomi78** [comment]
I wrote this to broadlink support and IHC app developer:

"Hi.
I recently realised that google home devices support broadkink skills. I can add it as an account google home center app and it works.
It was worked fine untill i changed my router. Now my google home mini and google home app not start the broadlink rm pro+...although all the scenes and lamp devices still recognised by home app. So all devices and scenes which i previously setup at IHC i see also at google home app.
Also ihc app work good standalone.

Only problem that when i want to activate broadlink rm pro+ through google home it not trigger rm pro+ to send IR or RF signal.
I do not want to delete rm pro from ihc app as all sub devices will also deleted.
Do you have a solution for this issue.
It seems the problem appears when i add the broadlink rm pro to another wifi router through the ihc app. After this the IHC works good again on the new access poimt (wifi) but the google cannot control it.
I need to reset and delete the main device from IHC and add all the subdiveces again to make it work again with google home.
If it possibele Im also want to save my sub devices settings from IHC..as it seems i need to delete the main device in case of a wifi change. And if i delete it it also clear all may remotes and sub devices.

Please help

Thanks "

**Comment by Ltomi78** [comment]
![screenshot_2018-02-05-12-48-19](https://user-images.githubusercontent.com/34459398/35803656-8053f4c2-0a6c-11e8-9639-e1fbb0d27268.png)
i v got reply froma boradlink support. They say google home and broadlink account works only with smart plugs yet :)
it is interesting...they say RM pro s are in docking phase

You can check this forum at the Google side 👍 
https://productforums.google.com/forum/#!topic/googlehome/Wp5TQZtYgS4;context-place=forum/googlehome

**Comment by Anthony1001windsurf** [comment]
Hi, 
RM Bridge and Broadlink pro3

2x pro versions 
Firmware 20020
and
Firmware v26 
work with the RM bridge without any issue.
Pro 3 (with holes for wall hanging) 
Firmware v52  is not recognised my rm bridge.
Having to source Pro 2 in order to perform Home Automation for IR and RF.
Have i missed any work arounds for Pro 3 and Rm bridge?

**Comment by Ltomi78** [comment]
No..it simply not works till broadlink not provide an SDK form rm bridge and rm plugin developers.
BUT we can now use the voice control with google home even the pro+ with v52...so i dont care as i want the rm bridge for voice operations

**Comment by Anthony1001windsurf** [comment]
Thanks
cannot use google as there is potential risk to accessing home auto system.
Many thanks
Anthony

**Comment by Ltomi78** [comment]
Based on Paul Hibbert video on youtube it works well with Alexa also....probably it is not risk free....but everything can be hacked

**Comment by jakaboers** [comment]
What risk?
If anything broadlink is the worst,  data goes through Chinese servers

**Comment by torinnguyen** [comment]
RMPlugin v2.4.8 has been released to support RMPro+ and SC1, and sweep RF mode too.

**Comment by hiato6** [comment]
Amazing! Any chance the RM Bridge to support RM Pro+ soon?

**Comment by BrckedmyRMpro** [comment]
Has anyone got any updates on recovery of a bricked RMpro+? I'm patiently waiting and hoping someone find out the answer to recovering these units that have been flagged with the earlier (wrong) firmware found on the RM Plugging website.

**Comment by hiato6** [comment]
Hey @beckyricha, appreciate your work on this. Any hopes of supporting RM Pro+ with your tool since it works with the RMPlugin?

**Comment by fgsaezs2** [comment]
RM Plugin 2.5.0 now find my RM PRO 2017 RM09....
Working Ok and http bridge too.

**Comment by hiato6** [comment]
No luck for Becky's method yet?

**Comment by hiato6** [comment]
Hi @beckyricha, do you know of a ST DTH that creates an AC device to control temperature, mode, etc, but all in one place? like this one for the TV? Thanks [https://community.smartthings.com/t/release-generic-tv-remote-device-handler-using-broadlink-rm-pro-and-mini-with-alexa-support/117384](url)

**Comment by b3n-0** [comment]
Nothing new on this to make it twork with RM Pro+?
I still can't register my codes there, which is kind of sad since it works great with the other controls...

**Comment by hieppo** [comment]
Hi, are there any updates to this issue? I never thought it impacted me since my bridge is working great for many years. I had RM Bridge installed (v1.3.3) on an old Galaxy SII. I am able to detect the device with its Chinese name and all. The firmware has been updated to 20028 and it still works on the Galaxy SII.
I recently tried to install RM Bridge v1.3.3 on a Motorola XT720 but RM Management does not detect the Broadlink device. I tried manually entering the MAC and type as RM2 (this and RM1 is the only types that will allow to be added manually). But manually adding it does not work still. I tried installing v1.3.0 and I was able to find an old v1.2.0 but all those versions still will not allow RM Management to detect the device.
As mentioned above, RM Plugin does work in detecting it. It detected with the same Chinese character name but the type is RM2+. I was able to copy the code from RM Management from my Galaxy SII into the test box of RM Plugin and it works the same way.
Something has changed in the Broadlink to not allow detection from RM Bridge. My only guess is the firmware has been updated somehow and does not play nice with RM Bridge. I can try to flash back to an older Broadlink firmware but where do I get that and what version will work or readily available to download.
I think the issue and solution is similar to the [Homebridge project](https://github.com/lprhodes/homebridge-broadlink-rm/issues/208). The app will need to accept the new type.

I have to convert all my URL over to RM Plugin but it is a hassle and also I have to purchase the app or renew subscription yearly when I already have a working solution with RM Bridge.
One of the other solution I was thinking is to copy the configuration for RM Bridge from the Galaxy SII over to the new phone. However, I have no idea where all the conf files are stored on the phone. I should be able to see it and copy it since both phone are rooted. I just need to know the location if there is one.

UPDATE: There must be something incompatible with the Motorola XT720 I am trying to run RM Bridge. I took another old Samsung SIIx and installed the latest RM Bridge and the Broadlink was detected and working. So I am not sure what's happening.
